internal/api/impl/auth/auth.go

```diff
@@ -141,3 +143,52 @@ func (e *endpoint) logout(ctx echo.Context) error {
 
 	return ctx.Redirect(302, "/")
 }
+
+// withOAuthErrorMdw applies to the handler a middleware that the error is well managed in oauth2.0 context.
+// This will transform any oauth2.0 retrieve token error into a bad request error, as suggested in the specs:
+// https://www.rfc-editor.org/rfc/rfc6749#section-5.2
+// https://www.rfc-editor.org/rfc/rfc8628#section-3.5
+// Otherwise, and only if the error is not an explicit Perses or HTTP error, we send back an opaque InternalError.
+func withOAuthErrorMdw(handler echo.HandlerFunc) echo.HandlerFunc {
```

Comment:
We can maybe put it directly in HandleError, I was not sure so I have it here now. To be discussed

Reply:
@Nexucis can you give me opinion on that ?

Reply:
yeah I think it's good to do it like that. I'm just proposing some changes to be more focus on oauth and oidc error management and let the generic `HandleError` taking care on all other errors